src/dragonball/src/dbs_virtio_devices/src/vhost/vhost_user/connection.rs

```diff
@@ -0,0 +1,560 @@
+// Copyright (C) 2019-2023 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Helper utilities for vhost-user communication channel.
+
+use std::ops::Deref;
+use std::os::unix::io::{AsRawFd, RawFd};
+
+use dbs_utils::epoll_manager::{EventOps, EventSet, Events};
+use dbs_utils::eventfd::EventFd;
+use log::*;
+use vhost_rs::vhost_user::message::{VhostUserProtocolFeatures, VhostUserVringAddrFlags};
+use vhost_rs::vhost_user::{
+    Error as VhostUserError, Listener as VhostUserListener, Master, VhostUserMaster,
+};
+use vhost_rs::{VhostBackend, VhostUserMemoryRegionInfo, VringConfigData};
+use virtio_queue::QueueT;
+use vm_memory::{
+    Address, GuestAddress, GuestAddressSpace, GuestMemory, GuestMemoryRegion, MemoryRegionAddress,
+};
+
+use super::super::super::device::VirtioDeviceConfig;
+use super::super::super::{Error as VirtIoError, Result as VirtIoResult};
```

Comment:
```suggestion
use super::super::super::{Error as VirtioError, Result as VirtioResult};
```

Reply:
It looks like other files already use VirtIoError and VirtIoResult.